backend/employee-payroll-service/src/main/java/com/mycompany/employee/payroll/service/exception/BaseServiceException.java

```diff
@@ -0,0 +1,20 @@
+package com.mycompany.employee.payroll.service.exception;
+
+import lombok.Getter;
+
+@Getter
+public class BaseServiceException extends RuntimeException {
```

Comment:
Since BaseServiceException act as foundation of service error. I think this class can use abstract class while create new exception to handle general error.

- BaseServiceException -> InternalServiceException
- BaseServiceException -> DataNotFoundException

So use it at ControllerAdvice case by case.